### PR TITLE
Centos7 (nginx|apache) & pg

### PR DESCRIPTION
**Pull request by atarkowska**
Travis should test centos7 installation
- [x] Travis deserves web_to_login script to fully test if services are really running. https://gist.github.com/aleksandra-tarkowska/9d5ed3033c3f9126c397
- [x] nginx
- [x] apache
- [x] make autogenerate.sh generic (remove hardcoded tokens)

NOTE: _autogenerate.sh was improved as previously it was not possible to add multiple tokens to skip docker workaround in `step01_centos7_pg_deps.sh`. New function looks for `if [ "${container:-}" = docker ]; then` or `if [ ! "${container:-}" = docker ]; then` and grab appropriate content automatically. Now we can have as many workarounds as we like. `remove_docker_workaround` can be reused across all steps if needed in the future._ cc: @jburel 

TEST the following for both PostgreSQL 9.4 and 9.5

```
$ cd linux/test
$ export ENV=centos7_nginx
```

```
$ PGVER=pg94 ./docker-build.sh $ENV
OSX: $ DMNAME=dev ./test_services.sh # docker machine can be obtained from docker-machine ls
UNIX: $ ./test_services.sh
# test PG service manually
OSX: $ docker run -d --privileged -p 8888:80 --name omeroinstall omero_install_test_$ENV
UNIX: $ docker run -d --name omeroinstall -v /sys/fs/cgroup:/sys/fs/cgroup:ro -v /run omero_install_test_$ENV
wait 10 sec
$ docker exec -it omeroinstall /bin/bash -c "service postgresql-9.4 status"
```

```
$ PGVER=pg95 ./docker-build.sh $ENV
OSX: $ DMNAME=dev ./test_services.sh # docker machine can be obtained from docker-machine ls
UNIX: $ ./test_services.sh# test PG service manually
# test PG service manually
OSX: $ docker run -d --privileged -p 8888:80 --name omeroinstall omero_install_test_$ENV
UNIX: $ docker run -d --name omeroinstall -v /sys/fs/cgroup:/sys/fs/cgroup:ro -v /run omero_install_test_$ENV
wait 10 sec
$ docker exec -it omeroinstall /bin/bash -c "service postgresql-9.5 status"
Redirecting to /bin/systemctl status  postgresql-9.5.service
● postgresql-9.5.service - PostgreSQL 9.5 database server
   Loaded: loaded (/usr/lib/systemd/system/postgresql-9.5.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2016-04-08 13:59:49 UTC; 792ms ago
  Process: 91 ExecStart=/usr/pgsql-9.5/bin/pg_ctl start -D ${PGDATA} -s -w -t 300 (code=exited, status=0/SUCCESS)
  Process: 63 ExecStartPre=/usr/pgsql-9.5/bin/postgresql95-check-db-dir ${PGDATA} (code=exited, status=0/SUCCESS)
 Main PID: 95 (postgres)
   CGroup: /system.slice/postgresql-9.5.service
           ├─ 95 /usr/pgsql-9.5/bin/postgres -D /var/lib/pgsql/9.5/data
           ├─104 postgres: logger process   
           ├─542 postgres: checkpointer process   
           ├─543 postgres: writer process   
           ├─544 postgres: wal writer process   
           ├─545 postgres: autovacuum launcher process   
           └─546 postgres: stats collector process   

Apr 08 13:59:43 34523d17c036 systemd[1]: Starting PostgreSQL 9.5 database server...
Apr 08 13:59:43 34523d17c036 pg_ctl[91]: pg_ctl: another server might be running; trying to start server anyway
Apr 08 13:59:43 34523d17c036 pg_ctl[91]: < 2016-04-08 13:59:43.541 UTC >LOG:  redirecting log output to logging collector process
Apr 08 13:59:43 34523d17c036 pg_ctl[91]: < 2016-04-08 13:59:43.541 UTC >HINT:  Future log output will appear in directory "pg_log".
Apr 08 13:59:49 34523d17c036 systemd[1]: Started PostgreSQL 9.5 database server.
```

```
$ export ENV=centos7_apache24
$ ./docker-build.sh $ENV
OSX: $ DMNAME=dev ./test_services.sh # docker machine can be obtained from docker-machine ls
UNIX: $ ./test_services.sh
```

nginx/apache version will appear in the curl header in the very end of test see travis for more details

TEST autogenerate.sh

```
$ git checkout develop
$ bash autogenerate.sh 
$ mv walkthrough_centos7.sh walkthrough_centos7_develop.sh
$ git checkout centos7-nginx
$ bash autogenerate.sh 
$ diff walkthrough_centos7.sh walkthrough_centos7_develop.sh
40d39
< 
43d41
< 
104d101
< mv /etc/nginx/conf.d/default.conf /etc/nginx/conf.d/default.disabled
108d104
< 
135d130
< 
142d136
< cp omero-systemd.service /etc/systemd/system/omero.service
143a138
> cp omero-systemd.service /etc/systemd/system/omero.service
```

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-DEV-merge-homebrew/ICE=3.5,OSX=snapper#148](https://ci.openmicroscopy.org/job/OMERO-DEV-merge-homebrew/ICE=3.5,OSX=snapper/148/). See the [console output](https://ci.openmicroscopy.org/job/OMERO-DEV-merge-homebrew/ICE=3.5,OSX=snapper/148/consoleText) for more details.
Possible conflicts:
- PR #95 aleksandra-tarkowska 'Redis'
  - linux/test/docker-build.sh

**Comment by atarkowska** [comment]
green ;-)

**Comment by joshmoore** [comment]
```
 Elapsed time 12 min 2 sec
 Total time 1 hr 42 min 25 sec
```

File: .travis.yml [hunk 1]
```diff
@@ -19,5 +21,7 @@ before_install:
   - sudo DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" docker-engine
 
 script:
-    - cd linux/test && ./docker-build.sh $ENV && docker run -d omero_install_test_$ENV
+    - cd linux/test
+    - ./docker-build.sh $ENV
+    - if [[ "$ENV" == "centos7" ]]; then ./test_services.sh; fi
```
Comment (joshmoore) [inline .travis.yml]: `test_services.sh` looks quite nice, but is now only being run on a subset, right?

Comment (atarkowska) [inline .travis.yml]: yes, it is not valid for the rest, more work needed

Comment (atarkowska) [inline .travis.yml]: I mostly concentrated on missing c7 travis. apache is now tested as well

**Comment by jburel** [comment]
The test script assumes that docker-machine is called `omerodev`, better to specify as an argument

**Comment by jburel** [comment]
I have fixed a problem in the autogenerate if os not centos7, you can cherry-pick the commit https://github.com/jburel/omero-install/commit/7b14ad4c88fdd9fbf179cdd6389df10d3df35d9e

**Comment by jburel** [comment]
I will remove from travis.yml the following.
`ENV=centos7_nginx PGVER=pg95` 
Or we add all the possible combinations for all os => a build will take a long time.

File: .travis.yml (hunk 1 above)
Comment (jburel) [inline .travis.yml]: We can expand it to other os in a follow-up PR.

**Comment by atarkowska** [comment]
thanks @jburel above should address all your comments. refresh the page to update PR description

**Comment by atarkowska** [comment]
travis testing matrix is already big, removed

**Comment by jburel** [comment]
Thanks for adjusting the test script, it works nicely now ;-)
I am a bit confused with the new section in the README. 
Are you thinking to add some of the lines you have in the description of this PR to the README?

**Comment by atarkowska** [comment]
sorry put that in the wrong file, see now :)

**Comment by jburel** [comment]
It will be better to move it under the existing `Testing CentOS 7` section.

**Comment by atarkowska** [comment]
sorry, missed that, now in c7 section

**Comment by jburel** [comment]
Thanks for moving it, we can adjust the content in another PR.
Merging so we can move forward

**Pull request merged**